test/cases/ostree-vsphere.sh

```diff
@@ -0,0 +1,884 @@
+    -e vsphere_vm_name="${DC70_VSPHERE_VM_NAME}" \
+    /usr/share/tests/osbuild-composer/ansible/vmdk-create-vm.yaml || RESULTS=0
+
+DC70_GUEST_ADDRESS=$(cat /usr/share/tests/osbuild-composer/ansible/vmdk_ip.txt)
+sudo rm -fr /usr/share/tests/osbuild-composer/ansible/vmdk_ip.txt
+greenprint "🛃 Edge VM IP address is: ${DC70_GUEST_ADDRESS}"
+
+# Run ansible check on edge vm
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+for _ in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up "${DC70_GUEST_ADDRESS}")"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# With new ostree-libs-2022.6-3, edge vm needs to reboot twice to make the /sysroot readonly
+sudo ssh "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" "${IGNITION_USER}@${DC70_GUEST_ADDRESS}" "echo ${IGNITION_USER_PASSWORD} |nohup sudo -S systemctl reboot &>/dev/null & exit"
+# Sleep 10 seconds here to make sure vm restarted already
+sleep 10
+for _ in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up "${DC70_GUEST_ADDRESS}")"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# Check image installation result
+check_result
+
+greenprint "🕹 Get ostree install commit value"
+INSTALL_HASH=$(curl "${PROD_REPO_URL}/refs/heads/${OSTREE_REF}")
+
+sudo tee "${TEMPDIR}"/inventory > /dev/null << EOF
+[ostree_guest]
+${DC70_GUEST_ADDRESS}
+
+[ostree_guest:vars]
+ansible_python_interpreter=/usr/bin/python3
+ansible_user=${IGNITION_USER}
+ansible_private_key_file=${SSH_KEY}
+ansible_ssh_common_args="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+ansible_become=yes
+ansible_become_method=sudo
+ansible_become_pass=${IGNITION_USER_PASSWORD}
+EOF
+
+# Test IoT/Edge OS
+sudo ansible-playbook -v -i "${TEMPDIR}"/inventory \
+    -e image_type=redhat \
+    -e ostree_commit="${INSTALL_HASH}" \
+    -e skip_rollback_test="true" \
+    -e edge_type=edge-vsphere \
+    -e fdo_credential="false" \
+    -e sysroot_ro="$SYSROOT_RO" \
+    /usr/share/tests/osbuild-composer/ansible/check_ostree.yaml || RESULTS=0
+check_result
+
+##################################################################
+##
+## Create vm on datacenter8.0 and test it
+##
+##################################################################
+# Vsphere setup
+VSPHERE_DC80="Datacenter8.0"
+VSPHERE_DC80_HOST="10.73.227.56"
+VSPHERE_DC80_DATASTORE="datastore-56"
+DC80_VSPHERE_VM_NAME="${COMPOSE_ID}-80"
+
+# Remove all existing disk files
+greenprint "🧹 Remove existing vmdk disk files in vsphere datacenter"
+sudo sshpass -p "${DATACENTER_PASSWORD}" ssh "${SSH_OPTIONS[@]}" "${DATACENTER_USERNAME}@${VSPHERE_DC80_HOST}" \
+    rm -fr /vmfs/volumes/${VSPHERE_DC80_DATASTORE}/edge-vsphere
+
+greenprint "📋 Uploading vmdk image to vsphere datacenter 8.0"
+export GOVC_DATACENTER="Datacenter8.0"
+govc import.vmdk -ds="datastore-56" -pool="/Datacenter8.0/host/Cluster8.0/Resources" ./edge-vsphere.vmdk
```

Comment:
Same comment.

Reply:
And you don't need copy vmdk image again! You can use `govc datastore.cp` to copy you uploaed vmdk file from Datacenter7.0-AMD to Datacenter8.0. That will save a lot of time.

Reply:
So you can cover more ESXi version, like 6.7.

Reply:
Changed to datastore.cp to clone disk to other datastore, it takes only a few minutes, thanks!